From 0707a1a86ca1fbb09b9788087c9af62343f0cb1c Mon Sep 17 00:00:00 2001
From: Hugues Fruchet <hugues.fruchet@foss.st.com>
Date: Tue, 24 Sep 2024 17:01:10 +0200
Subject: [PATCH] kmsallocator: fix stride with planar formats

This fixes a regression introduced by 37f2d2ac42b4e8033291742b17d304cd46fc66c46c4f52ea20ce27730501348c6e575b8d35580239

Fixes #3801

Upstream-Status: Backport

https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/7566
---
 sys/kms/gstkmsallocator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sys/kms/gstkmsallocator.c b/sys/kms/gstkmsallocator.c
index 3ec2fdb32d..84359b0782 100644
--- a/sys/kms/gstkmsallocator.c
+++ b/sys/kms/gstkmsallocator.c
@@ -122,7 +122,7 @@ gst_kms_allocator_memory_create (GstKMSAllocator * allocator,
     /* Overwrite the video info's stride and offset using the pitch calculcated
      * by the kms driver. */
     stride = gst_video_format_info_extrapolate_stride (vinfo->finfo, i, pitch);
-    GST_VIDEO_INFO_PLANE_STRIDE (vinfo, i) = pitch;
+    GST_VIDEO_INFO_PLANE_STRIDE (vinfo, i) = stride;
     GST_VIDEO_INFO_PLANE_OFFSET (vinfo, i) = offs;
 
     /* Note that we cannot negotiate special padding betweem each planes,
-- 
2.25.1

